src/game/client.cpp

```diff
@@ -1482,6 +1482,142 @@ namespace client
     }
     //ICOMMAND(0, sendmap, "", (), sendmap());
 
+    static void startcmd(char *map, char *mode_str, char *muts_str)
+    {
+        static const struct {const char *name; int val;} mode_names[] =
+        {
+            {"demo",       G_DEMO},
+            {"edit",       G_EDITMODE},
+            {"deathmatch", G_DEATHMATCH},
+            {"dm",         G_DEATHMATCH},
+            {"capture",    G_CAPTURE},
+            {"ctf",        G_CAPTURE},
+            {"defend",     G_DEFEND},
+            {"dac",        G_DEFEND},
+            {"bomber",     G_BOMBER},
+            {"bb",         G_BOMBER},
+            {"race",       G_RACE},
+        };
+        static const struct {const char *name; int val; int modes;} mut_names[] =
```

Comment:
This should not be a struct array. Come on, we've got the power of C++ 11/14. Why not make it an `std::set<std::string, std::pair<size_t, size_t>>` or so? There is really no reason to use anonymous structs.

Reply:
Why not use a struct array? I like it when code is simple and easy to understand. Why should I put a wrapper around my char array (called std::string) that might allocate it on the heap and do some other undefined things when I don't need that? And why should I do similar thing with the whole array? It might even be slower. I never add things to these arrays either so they don't need to grow. They are just static data.